Scripts/BuildDatabank/AddData.py

```diff
@@ -16,572 +35,550 @@
+                                        "REQUIRED": is_required,
+                                        "SIZE_MB": file_size_mb,
+                                        "HASH": file_hash,
+                                    }
+                                ]
+                            ),
+                        ],
+                        ignore_index=True,
+                    )
+                    files_list.append([file_provided[0], file_hash])
+
+                    # Find the keys of the required files to calculate the master_hash
+                    if is_required:
+                        sha1_list_requied.append(file_hash)
+
+                    sim_hashes[key_sim] = files_list  # TODO Problematic
+        except KeyError as e:  # It is notmal that fails for "ID" and "SOFTWARE"
+            continue
+
+    logger.info(f"Summary of downloaded files:{os.linesep}")
+    print(df_files)
+    print()
+
+    # Calculate the hash of a file contaning the hashes of each of the required files
+    # This should be always invariant as it will be used unique identifier for a simualtion
+    # Note order the hashes of the required files before calculating the hash (That means that the required files cannot change)
+
+    # Calculates numbers of lipid molecules in each leaflet. This is done by checking on which side of the centre
+    # of mass the membrane each the centre of mass of a lipid molecule is.
+    # If a lipid molecule is split so that headgroup and tails are their own residues, the centre of mass of the
+    # headgroup is used in the calculation.
+    ################################################################################################################
 
-## Check that the number of atoms between data and README.yaml match
+    logger.info(
+        "Calculating the numbers of lipid molecules in each leaflet based on the center of mass of the membrane and lipids."
+    )
+    logger.info(
+        "If a lipid molecule is split to multiple residues, the centre of mass of the headgroup is used."
+    )
 
-number_of_atomsTRJ = len(u.atoms)
+    top = ""
+    traj = ""
 
-number_of_atoms = 0
-for key_mol in all_molecules:
-    mapping_dict = {}
-    try:
-        mapping_file = os.path.join(
-            os.getcwd(), "mapping_files", sim["COMPOSITION"][key_mol]["MAPPING"]
-        )
-    except:
-        continue
+    if sim["SOFTWARE"] == "gromacs":
+        top = os.path.join(dir_tmp, sim["TPR"][0][0])
+        traj = os.path.join(dir_tmp, sim["TRJ"][0][0])
+    elif sim["SOFTWARE"] == "openMM" or sim["SOFTWARE"] == "NAMD":
+        traj = os.path.join(dir_tmp, sim["TRJ"][0][0])
+        top = os.path.join(dir_tmp, sim["PDB"][0][0])
     else:
-        with open(mapping_file, "r") as yaml_file:
-            mapping_dict = yaml.load(yaml_file, Loader=yaml.FullLoader)
-        yaml_file.close()
+        logger.error(
+        "SOFTWARE '%s' is not a proper option.\n"
+        "Use either 'gromacs', 'openMM', or 'NAMD'.")
```

Comment:
Someone should write test case for openMM and NAMD data and he should not be me. Otherwise I would consider that they don't work. And they probably don't even before this PR. I think NAMD analysis I did tested somewhere.